cluster/v1/types.go

```diff
@@ -170,3 +170,52 @@ type ManagedClusterList struct {
 	// Items is a list of managed cluster.
 	Items []ManagedCluster `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Cluster"
+
+// ManagedClusterSet is a set of managed clusters.
+type ManagedClusterSet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of the desired managed clusters
+	Spec ManagedClusterSetSpec `json:"spec"`
+}
+
+// ManagedClusterSetSpec describes the attributes of the desired managed clusters
+type ManagedClusterSetSpec struct {
+	// ClusterSelector represents a selector to select a set of managed clusters
+	// If empty, the cluster set will include all managed clusters
+	// +optional
+	ClusterSelector ClusterSelector `json:"clusterSelector,omitempty"`
```

Comment:
I think you want to allow a slice of these so that ORs are easy to express.

Reply:
Updated.